### PR TITLE
Feature: Groq updated for 0.9

### PR DESCRIPTION
**Pull request by Gustolandia**
_Before you submit a pull request, please make sure you have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md)._

**This pull request is related to:**

- [ ] A bug
- [x ] A new feature
- [ ] Documentation
- [ ] Other (please specify)

**I have checked the following:**

- [ ] I have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md);
- [ ] I have added new tests (for bug fixes/features);
- [ ] I have added/updated the documentation (for bug fixes / features).

**Description:**
Please provide a clear and concise description of the changes you are proposing here.

**Related issues:**
Please, link the related issues that this pull request will resolve here (if any).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,5 +21,8 @@
     "lerna": "^8.1.2",
     "lint-staged": "^15.2.2",
     "prettier": "^3.2.5"
+  },
+  "dependencies": {
+    "uuid": "^11.0.2"
```
Comment (cabljac) [inline package.json:29]: If this is just needed for tests it should be a devDependency, something like this:

```
npm uninstall uuid
```

and
```
npm install -D uuid
```

Comment (Gustolandia) [reply package.json:29]: Changed

File: plugins/groq/src/index.ts [hunk 2]
```diff
@@ -14,69 +14,63 @@
...
+   * Base URL for overriding the default API endpoint, useful for targeting specific
+   * environments or versions. Defaults to process.env['GROQ_BASE_URL'] if unspecified.
    */
   baseURL?: string | null | undefined;
 
   /**
-   * The maximum amount of time (in milliseconds) that the client should wait for a response
-   * from the server before timing out a single request.
-   *
-   * Note that request timeouts are retried by default, so in a worst-case scenario you may wait
```
Comment (cabljac) [inline plugins/groq/src/index.ts]: Can we keep their comments present? We shouldn't remove this note unless it's not relevant any more

File: plugins/groq/src/index.ts [hunk 3]
```diff
@@ -14,69 +14,63 @@
...
+   * Timeout value (in milliseconds) for waiting on a response from the API.
+   * Provides control over network delays and retries in long requests.
    */
   timeout?: number;
 
   /**
-   * The maximum number of times that the client will retry a request in case of a
-   * temporary failure, like a network error or a 5XX error from the server.
-   *
-   * @default 2
```
Comment (cabljac) [inline plugins/groq/src/index.ts]: Again keep their comments the same, I think `@default` is a jsDoc directive/tag, and we shouldn't change this

File: plugins/groq/src/index.ts [hunk 4]
```diff
@@ -14,69 +14,63 @@
...
-    };
-  }
-);
+/**
+ * Initializes and returns the Groq plugin with provided or default options.
+ *
+ * @param options - Optional configuration settings for the plugin.
+ * @returns An object containing the models initialized with the Groq client.
+ */
+export const groq = async (options?: PluginOptions) => {
```
Comment (cabljac) [inline plugins/groq/src/index.ts]: We need to call genkitPlugin still here.

See this as an example of the new signature:

https://github.com/invertase/thefireco-genkit-plugins/blob/%40invertase/genkit-0.9/plugins/anthropic/src/index.ts#L47

File: plugins/groq/tests/groq_test.ts [hunk 5]
```diff
@@ -121,51 +63,34 @@ describe('toGroqRequestBody', () => {
...
       top_logprobs: 10,
       user: 'exampleUser123',
-      response_format: {
-        type: 'text',
-      },
+      response_format: { type: 'text' },
     };
+
     const actualOutput = toGroqRequestBody('llama-3-8b', request);
-    console.log(`actualOutput.stop: ${actualOutput.stop}`);
```
Comment (cabljac) [inline plugins/groq/tests/groq_test.ts:167]: This is a nitpick tbf - 

If these were our tests i'd say remove the log, but we should keep changes minimal if we're just bumping genkit. I'd put this log back in

File: plugins/groq/tests/groq_test.ts [hunk 6]
```diff
@@ -33,78 +27,26 @@ describe('toGroqRole', () => {
...
-        {
-          toolResponse: {
-            ref: 'ref123',
-            name: 'getResponse',
-            output: 'Sample response',
-          },
-        },
-      ],
-    },
+    { role: 'user', content: [{ text: 'Hello, world!' }] },
```
Comment (cabljac) [inline plugins/groq/tests/groq_test.ts:37]: Can you run `npm run format` from the root directory? I think that's why we're seeing format changes here

File: plugins/groq/src/index.ts (hunk 2 above)
Comment (cabljac) [reply plugins/groq/src/index.ts]: I would have a look through and try not to change code comments unless it's absolutely necessary here (when we put this through to them we dont want them to complain we've changed the comments without reason)

File: plugins/groq/src/index.ts [hunk 7]
```diff
@@ -14,69 +14,63 @@
...
-  llama3x70b,
-  llama3x8b,
-  gemma7b,
-  mixtral8x7b,
-  groqModel,
-  SUPPORTED_GROQ_MODELS,
-} from './groq_models';
+import { groqModel, SUPPORTED_GROQ_MODELS } from './groq_models';
 
-export { llama3x70b, llama3x8b, gemma7b, mixtral8x7b };
```
Comment (cabljac) [inline plugins/groq/src/index.ts:28]: We should still export these.

File: plugins/groq/src/index.ts [hunk 8]
```diff
@@ -14,69 +14,63 @@
...
+// Exporting SUPPORTED_GROQ_MODELS for external modules to access available models easily
+export { SUPPORTED_GROQ_MODELS };
```
Comment (cabljac) [inline plugins/groq/src/index.ts]: We shouldn't export this if the plugin didn't already

**Review by cabljac — CHANGES_REQUESTED** [review]

File: plugins/groq/src/index.ts (hunk 7 above)
Comment (Gustolandia) [reply plugins/groq/src/index.ts:28]: Changed

File: plugins/groq/src/index.ts (hunk 8 above)
Comment (Gustolandia) [reply plugins/groq/src/index.ts]: Done

File: plugins/groq/src/index.ts (hunk 2 above)
Comment (Gustolandia) [reply plugins/groq/src/index.ts]: Done

File: plugins/groq/src/index.ts (hunk 3 above)
Comment (Gustolandia) [reply plugins/groq/src/index.ts]: Done

File: plugins/groq/src/index.ts (hunk 4 above)
Comment (Gustolandia) [reply plugins/groq/src/index.ts]: Done, thanks

File: plugins/groq/tests/groq_test.ts (hunk 6 above)
Comment (Gustolandia) [reply plugins/groq/tests/groq_test.ts:37]: Done.

File: plugins/groq/tests/groq_test.ts (hunk 5 above)
Comment (Gustolandia) [reply plugins/groq/tests/groq_test.ts:167]: Done.

File: plugins/groq/package.json [hunk 9]
```diff
@@ -24,16 +24,17 @@
...
-    "zod": "^3.22.4"
+    "groq-sdk": "^0.3.3"
   },
   "peerDependencies": {
-    "@genkit-ai/ai": "^0.5.0",
-    "@genkit-ai/core": "^0.5.0"
+    "genkit": "^0.9.0-dev.2"
   },
   "devDependencies": {
+    "@types/hast": "^3.0.4",
```
Comment (cabljac) [inline plugins/groq/package.json:33]: What do we use these types for? It's fine if we need them, just wondering

File: plugins/groq/tests/groq_test.ts [hunk 10]
```diff
@@ -33,78 +27,26 @@ describe('toGroqRole', () => {
...
-        role: 'assistant',
-        content: 'How can I assist you today?',
-      },
-      {
-        role: 'assistant',
-        tool_call_id: 'ref123',
-        content: 'Sample response',
-      },
+      { role: 'user', content: 'Hello, world!' },
+      { role: 'assistant', content: 'How can I assist you today?' },
```
Comment (cabljac) [inline plugins/groq/tests/groq_test.ts:44]: Did you run `npm format` in the root

File: plugins/groq/tests/groq_test.ts [hunk 11]
```diff
@@ -121,51 +63,35 @@ describe('toGroqRequestBody', () => {
...
-      response_format: {
-        type: 'text',
-      },
+      response_format: { type: 'text' },
     };
+
     const actualOutput = toGroqRequestBody('llama-3-8b', request);
     console.log(`actualOutput.stop: ${actualOutput.stop}`);
-    assert.deepStrictEqual(actualOutput, expectedOutput);
+    assert.deepStrictEqual(
```
Comment (cabljac) [inline plugins/groq/tests/groq_test.ts:91]: how come we have to add a JSON parse here? I dont think it's an issue, just wondering

File: plugins/groq/package.json [hunk 12]
```diff
@@ -57,9 +58,9 @@
   "scripts": {
     "check": "tsc",
     "compile": "tsup-node",
-    "build:clean": "rm -rf ./lib",
+    "build:clean": "rimraf ./lib",
     "build": "npm-run-all build:clean check compile",
     "build:watch": "tsup-node --watch",
-    "test": "node --import tsx --test ./tests/*_test.ts"
+    "test": "tsx \"./tests/groq_test.ts\""
```
Comment (cabljac) [inline plugins/groq/package.json]: Can we keep the test script generic? Now it just runs the one test file

**Review by cabljac — CHANGES_REQUESTED** [review]
mostly lgtm! Couple of comments and questions

File: plugins/groq/tests/groq_test.ts (hunk 10 above)
Comment (Gustolandia) [reply plugins/groq/tests/groq_test.ts:44]: Yes, I did. From the root.

File: plugins/groq/package.json (hunk 12 above)
Comment (Gustolandia) [reply plugins/groq/package.json]: I was having problems running it in my PC. That is why I did that I will try to do a generic one.

File: plugins/groq/package.json (hunk 9 above)
Comment (Gustolandia) [reply plugins/groq/package.json:33]: I had an error in tsconfig.json without them

File: plugins/groq/tests/groq_test.ts (hunk 11 above)
Comment (Gustolandia) [reply plugins/groq/tests/groq_test.ts:91]: To remove undefined fields (I put a comment about it)

File: plugins/groq/package.json (hunk 12 above)
Comment (Gustolandia) [reply plugins/groq/package.json]: I have made a generic version. I will commit the change.

**Review by cabljac — APPROVED** [review]
lgtm!

**Pull request merged**